src/main/java/com/fouristhenumber/utilitiesinexcess/common/items/ItemInversionSigilActive.java

```diff
@@ -2,29 +2,527 @@
+            if (world.getBlock(x + REDSTONE_SPOTS_ALTERNATE[i][0], y, z + REDSTONE_SPOTS_ALTERNATE[i][1])
+                != Blocks.redstone_wire) {
+                alternateTrue = false;
+                break;
+            }
+            if (world.getBlock(x + STRING_SPOTS_ALTERNATE[i][0], y, z + STRING_SPOTS_ALTERNATE[i][1])
+                != Blocks.tripwire) {
+                alternateTrue = false;
+                break;
+            }
+        }
+        return baseTrue || alternateTrue;
+    }
+
+    private void siegeStart(World world, int beaconX, int beaconY, int beaconZ, EntityPlayer player) {
+        EntitySiegeProperty source = getProperties(player);
+        source.beaconSpawnX = beaconX;
+        source.beaconSpawnY = beaconY;
+        source.beaconSpawnZ = beaconZ;
+        source.siege = true;
+        source.siegeMobsKilled = 0;
+        source.siegeTimer = 0;
+        for (Entity curentity : player.getEntityWorld().loadedEntityList) {
+            if (curentity instanceof EntityEnderman) {
+                curentity.setDead();
+            }
+        }
+        for (int i = 0; i < 5; i++) {
+            EntityLightningBolt bolt = new EntityLightningBolt(
+                world,
+                beaconX + BOLT_POSITIONS[i][0] + 0.5,
+                beaconY + 0.5,
+                beaconZ + BOLT_POSITIONS[i][1] + 0.5);
+            world.addWeatherEffect(bolt);
+            world.setBlock(beaconX + BOLT_POSITIONS[i][0], beaconY, beaconZ + BOLT_POSITIONS[i][1], Blocks.air);
+        }
+    }
+
+    private void siegeEnd(boolean Won, EntityPlayer player) {
+        EntitySiegeProperty source = getProperties(player);
+        source.siege = false;
+        source.siegeMobsKilled = 0;
+        source.siegeTimer = 0;
+        for (Entity curentity : player.getEntityWorld().loadedEntityList) {
+            if (curentity instanceof EntityMob) {
+                curentity.setDead();
+            }
+        }
+        if (Won) {
+            for (int i = 0; i < player.inventory.getSizeInventory(); i++) {
+                ItemStack is = player.inventory.getStackInSlot(i);
+                if (is != null && is.getItem() == this) {
+                    player.inventory
+                        .setInventorySlotContents(i, new ItemStack(ModItems.PSEUDO_INVERSION_SIGIL.get(), 1));
+                    break;
+                }
+            }
+        }
+    }
+
+    private boolean checkChest(TileEntityChest chest, ItemStack[] CHECKED_ITEMS, int ITEM_REQUIREMENT) {
+        int CHECKED_ITEMS_SIZE = CHECKED_ITEMS.length;
+        int requiredItemsAmount = 0;
+        boolean[] hasItem = new boolean[CHECKED_ITEMS_SIZE];
+        for (int i = 0; i < chest.getSizeInventory(); i++) {
+            ItemStack stack = chest.getStackInSlot(i);
+            for (int j = 0; j < CHECKED_ITEMS_SIZE; j++) {
+                if (stack != null && ItemStack.areItemStacksEqual(stack, CHECKED_ITEMS[j])) {
+                    hasItem[j] = true;
```

Comment:
just straight up count amount here: `requiredItemsAmount++`, no need for an additional array and a loop